threescale_api/resources.py

```diff
@@ -1096,3 +1130,28 @@ def proxy(self) -> 'Proxy':
     @property
     def service(self) -> 'Service':
         return self.proxy.service
+
+
+class ProviderAccount(DefaultStateResource):
+    def __init__(self, entity_name='username', **kwargs):
+        super().__init__(entity_name=entity_name, **kwargs)
+
+    def set_role_member(self,  **kwargs):
+        log.info("Changes the role of the user of the provider account to member")
+        return self.set_state(state='member', **kwargs)
+
+    def set_role_admin(self, **kwargs):
+        log.info("Changes the role of the provider account to admin")
+        return self.set_state(state='admin', **kwargs)
+
+    def suspend(self, **kwargs):
+        log.info("Changes the state of the user of the provider account to suspended")
+        return self.set_state(tate='suspend', **kwargs)
+
+    def unsuspend(self, **kwargs):
+        log.info("Revokes the suspension of a user of the provider account")
+        return self.set_state(state='unsuspend', **kwargs)
+
+    def activate(self, **kwargs):
+        log.info("Changes the state of the user of the provider account to active")
+        return self.set_state(state='activate', **kwargs)
```

Comment:
Does this use the special endpoints that are used for this or will this just send the entire entity in update request?

Reply:
It uses its special endpoints:
![image](https://user-images.githubusercontent.com/7500621/98667498-bc63a400-234e-11eb-9113-193e37d9888d.png)

Reply:
Ah thanks!